src/plugins/discover/public/application/main/utils/fetch_all.ts

```diff
@@ -139,10 +141,13 @@ export function fetchAll(
             result: docs.length,
           });
         }
+        const dataView = searchSource.getField('index')!;
+
+        const resultDocs = docs.map((doc) => buildDataTableRecord(doc as EsHitRecord, dataView));
```

Comment:
that's the significant change, no longer feed the `dataSubjects.documents$`with raw data, flatten upfront, add unique ID